doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1906,9 +1906,38 @@ Numbers and strings
 
 Numbers and strings have no predefined semantics in the calculus. They are
 merely notations that can be bound to objects through the notation mechanism.
-Initially, numbers are bound to Peano’s representation of natural
+Initially, numbers are bound to :n:`nat`, Peano’s representation of natural
 numbers (see :ref:`datatypes`).
 
+Valid Ranges for Numbers
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, numbers are interpreted as :n:`nat`\s, which is a unary representation.
+For example, :n:`3` is represented as `(S (S (S O)))`.  Computing with large :n:`nat`\s
+can lead to stack overflows.  You can explicitly specify a different interpretation,
+which will avoid this problem.  For example :n:`3%Z` represents the number as an integer.
+See :ref:`Scopes` and :n:`@term_scope`.
+
+.. todo PR: what is the specific type for the integer?  Is there a range limitation?
```

Comment:
What do you mean by "type" here?  What's your model of how this all works?

Reply:
"type" has the standard definition.  So `Z` is the common type for an integer and natural numbers as efficient bignums are `N`:

![image](https://user-images.githubusercontent.com/1253341/226386153-536eb3b9-b3e8-485f-8daf-b8373e0d79a4.png)